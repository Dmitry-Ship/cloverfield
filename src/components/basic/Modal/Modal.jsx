import React, { PropTypes } from 'react';

import { wrapper, content } from './Modal.scss';

const Modal = ({ children, closeModal }) => {
  const stop = e => e.stopPropagation();

  return (
    <div className={wrapper} onClick={closeModal} >
      <div className={content} onClick={stop} >
        {children}
      </div>
    </div>
  );
};

export default Modal;

Modal.propTypes = {
  closeModal: PropTypes.func.isRequired,
};
