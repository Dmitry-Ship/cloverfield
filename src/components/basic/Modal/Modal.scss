
.wrapper {
  position: fixed;
  display: flex;
  flex-direction: column;
  justify-content: center;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width:100%;
  height: 100vh;
  z-index: 2;
  background: rgba(black, 0.6);

}

.content {
  position: absolute;
  width: 100%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
