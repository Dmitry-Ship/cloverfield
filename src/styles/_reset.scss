body {
  font-family: Raleway;
}
  
a {
  text-decoration: none;
}

ul {
  list-style-type: none;
}

input, button, textarea {
  outline: none;
}

* {
  box-sizing: border-box;
}
