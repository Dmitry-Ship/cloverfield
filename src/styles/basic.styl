body
  font-family Rubik

a
  text-decoration none

ul
  list-style-type none

input, button, textarea
  outline none
*
  box-sizing border-box
