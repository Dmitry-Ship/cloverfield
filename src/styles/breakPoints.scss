@value mobile: (max-width: 768px);
@value laptop: (max-width: 1024px);
