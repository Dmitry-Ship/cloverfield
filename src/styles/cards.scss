.formCard {
  background: white;
  margin: auto;
  width: 600px;
  padding: 5rem 5rem;
  position: relative;
  composes: level-1 from './shadows.scss';
}

.infoCard {
  background: white;
  margin: auto;
  width: 600px;
  padding: 5rem 5rem;
  composes: level-1 from './shadows.scss';
}
@media only screen and (max-width: 768px) {
  .formCard {
    max-width: 95%;
    padding: 2.5rem 2.5rem;
  }
}

@media only screen and (max-width: 768px) {
  .infoCard {
    max-width: 90%;
    padding: 2.5rem 2.5rem;
  }
}
