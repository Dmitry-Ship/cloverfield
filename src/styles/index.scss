@import '_normalize.scss';
@import '_basic.scss';