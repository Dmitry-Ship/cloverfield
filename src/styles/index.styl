@import 'basic.styl'
@import 'mixins.styl'
