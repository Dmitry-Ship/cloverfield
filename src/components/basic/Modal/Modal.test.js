import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import styles from './Modal.scss';
import Modal from './Modal';

describe('<Modal />', () => {
  const closeModal = jest.fn();

  it('should render correctly', () => {
    const tree = renderer.create(
      <Modal closeModal={closeModal} >Hello World</Modal>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should simulate click on icon', () => {
    const wrapper = shallow(
      <Modal closeModal={closeModal} >Hello World</Modal>
    );

    wrapper.find('Icon').simulate('click');

    expect(closeModal).toHaveBeenCalled();
  });

  it('should simulate click on wrapper', () => {
    const closeModal = jest.fn();
    const wrapper = mount(
      <Modal closeModal={closeModal} >Hello World</Modal>
    );

    wrapper.find(`div.${styles.wrapper}`).simulate('click');

    expect(closeModal).toHaveBeenCalled();
  });
});
