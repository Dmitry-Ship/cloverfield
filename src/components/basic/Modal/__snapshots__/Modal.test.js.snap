// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render correctly 1`] = `
<div
  className="wrapper"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <i
      className="icon material-icons"
      onClick={[Function]}
    >
      close
    </i>
    Hello World
  </div>
</div>
`;
